TaskLayer/SearchTask/PostSearchAnalysisTask.cs

```diff
@@ -47,14 +47,17 @@ public MyTaskResults Run()
             }
 
             //update all psms with peptide info
-            Parameters.AllPsms = Parameters.AllPsms.Where(psm => psm != null).ToList();
-            Parameters.AllPsms.ForEach(psm => psm.ResolveAllAmbiguities());
+            if (Parameters.SearchParameters.SearchType != SearchType.NonSpecific) //if it hasn't been done already
```

Comment:
I don't really like that q-values are calculated in the nonspecific search without using the FdrAnalysisEngine. it's a code duplication thing. can we edit FdrAnalysisEngine to do FDR for each FDR category instead of having a separate duplicated FDR calculation method in NonSpecificEnzymeSearchEngine?

Reply:
The nonspecific method uses the fdranalysis engine for each category. A separate method is needed to aggregate them, though, which is what this does.